src/rgw/driver/rados/rgw_trim_bilog.cc

```diff
@@ -1143,11 +1307,77 @@ int BucketTrimCR::operate(const DoutPrefixProvider *dpp)
       }
     }
 
+    // check for orphaned objects in buckets. if there are objects add the bucket to the omap list
+    // "deleted_buckets" object.
+
+    using ReadCR = RGWSimpleRadosReadCR<DeletedBucketList>;
```

Comment:
BucketTrimCR runs on a fixed interval of ~20 minutes. i don't think we want to pay for all of this every time. as this omap list of deleted buckets grows, we'll have less time to spend on the actual trimming

i think it's best to leave this cleanup for the admin

Reply:
agreed

Reply:
@cbodley to prevent the omap object from growing too big, I was hoping that every time the command is run, we remove entries that don't have objects in them anymore . for that to happen, I tried listing those buckets in rgw_admin [L7397](https://github.com/ceph/ceph/pull/53799/commits/da328c2a2d08b636445398f8753bfd7ede735113#diff-f3b81fb31115c2616085efcfd6a8daed67a94d1041a03af5de48cc572978f547R7397). but we remove the bucket index objects as part of trimming in `BucketCleanIndexCollectCR()` . so any listing after that call would fail.

Reply:
please review 92dc547 and da328c2a2

Reply:
@cbodley could you please take a look at 461957aff008827c77621468c8b657132588d7b6 https://github.com/ceph/ceph/commit/92dc5474248d6c2f2cb0dcf696048f98d9e035d4 and https://github.com/ceph/ceph/commit/da328c2a2d08b636445398f8753bfd7ede735113